azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageMgmtCommonDao.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.daos;
+
+import azkaban.imagemgmt.dto.DeleteResponse;
+import azkaban.imagemgmt.exception.ImageMgmtException;
+
+/**
+ * This interface provides API for doing all the common operations for image management. Delete
+ * image version deletes image version and all the associated metadata.
+ */
+public interface ImageMgmtCommonDao {
+
+  /**
+   * Method to delete image version and associated metadata.
+   *
+   * @param imageTypeName
+   * @param versionId
+   * @param forceDelete
+   * @return
+   * @throws ImageMgmtException
+   */
+  public DeleteResponse deleteImageVersion(final String imageTypeName, final int versionId,
```

Comment:
Why is this method not part of ImageVersionDao?

Reply:
The purpose of delete image version is to delete complete metadata. Delete image version touches multiple tables such as image_versions, image_types, image_rampup_plan and image_rampup and hence required to use some of the existing daos. That is the reason why this method is part of the common dao.